optuna/pruners/threshold.py

```diff
@@ -94,7 +95,7 @@ def __init__(
         self._n_warmup_steps = n_warmup_steps
         self._interval_steps = interval_steps
 
-    def prune(self, study: Study, trial: FrozenTrial) -> bool:
+    def prune(self, study: 'optuna.study.Study', trial: FrozenTrial) -> bool:
```

Comment:
To use type hint, it needs `optuna.study.Study`.
However, `from optuna.study import Study` is a circular import and
`import optuna; optuna.study.Study` fails with a message `AttributeError: module 'optuna' has no attribute 'study'.`. (same as https://github.com/optuna/optuna/pull/809#discussion_r361407060)

As a workaround, I use a string literal.
(I want some suggestions if it is not good...)